src/psyclone/psyir/symbols/symbol.py

```diff
@@ -119,5 +257,68 @@ def visibility(self):
         '''
         return self._visibility
 
+    @property
+    def interface(self):
+        '''
+        :returns: the an object describing the interface to this Symbol.
+        :rtype: Sub-class of \
+            :py:class:`psyclone.psyir.symbols.symbol.SymbolInterface`
+        '''
+        return self._interface
+
+    @interface.setter
+    def interface(self, value):
+        '''
+        Setter for the Interface associated with this Symbol.
+
+        :param value: an Interface object describing how the Symbol is \
+                      accessed.
+        :type value: Sub-class of \
+            :py:class:`psyclone.psyir.symbols.symbol.SymbolInterface`
+
+        :raises TypeError: if the supplied `value` is of the wrong type.
+        '''
+        if not isinstance(value, SymbolInterface):
+            raise TypeError("The interface to a Symbol must be a "
+                            "SymbolInterface but got '{0}'".
+                            format(type(value).__name__))
+        self._interface = value
+
+    @property
+    def is_local(self):
+        '''
+        :returns: whether the Symbol has a Local interface.
+        :rtype: bool
+
+        '''
+        return isinstance(self._interface, LocalInterface)
+
+    @property
+    def is_global(self):
+        '''
+        :returns: whether the Symbol has a Global interface.
+        :rtype: bool
+
+        '''
+        return isinstance(self._interface, GlobalInterface)
+
+    @property
+    def is_argument(self):
+        '''
+        :returns: whether the Symbol has an Argument interface.
+        :rtype: bool
+
+        '''
+        return isinstance(self._interface, ArgumentInterface)
+
+    @property
+    def unresolved_interface(self):
```

Comment:
Can we rename this property to 'is_unresolved' to be consistent with the other interface checking methods?

Reply:
I agree. I was just keeping it the same as before (as this is just a copy of the code that was in DataSymbol). Changed.